onnxruntime/test/shared_lib/test_inference.cc

```diff
@@ -2508,11 +2515,13 @@ TEST(CApiTest, TestConfigureTensorRTProviderOptions) {
 
   const char* engine_cache_path = "./trt_engine_folder";
 
-  std::vector<const char*> keys{"device_id", "trt_fp16_enable", "trt_int8_enable", "trt_engine_cache_enable", "trt_engine_cache_path"};
+  std::vector<const char*> keys{"device_id", "trt_fp16_enable", "trt_int8_enable", "trt_engine_cache_enable",
+                                "trt_engine_cache_path", option_name.c_str()};
 
-  std::vector<const char*> values{"0", "1", "0", "1", engine_cache_path};
+  std::vector<const char*> values{"0", "1", "0", "1",
+                                  engine_cache_path, option_value.c_str()};
 
-  ASSERT_TRUE(api.UpdateTensorRTProviderOptions(rel_trt_options.get(), keys.data(), values.data(), 5) == nullptr);
+  ASSERT_TRUE(api.UpdateTensorRTProviderOptions(rel_trt_options.get(), keys.data(), values.data(), keys.size()) == nullptr);
```

Comment:
Here is a minor suggestion. 
Once you call api.UpdateTensorRTProviderOptions(), you might want to check the whether the provider options are being set simply through checking the returnin string from api.GetTensorRTProviderOptionsAsString().

You can reference the code `ASSERT_TRUE(s.find(engine_cache_path) != std::string::npos)` below and doing the same for trt_tactic_sources=-CUDNN,+CUBLAS ?

Reply:
Also while writing that test i discovered some typo's on my end so it was a good idea to do it 👍